content/docs/04-ai-sdk-ui/02-chatbot.mdx

```diff
@@ -634,6 +634,77 @@ messages.map(message => (
 ));
 ```
 
+## Resumable Streams (Experimental)
+
+The `useChat` hook has experimental support for resuming an ongoing chat generation stream by any client, either after a network disconnect or reloading the chat. This can be useful for building applications that involve long-running conversations or for ensuring that messages are not lost in case of network failures.
+
+The following are the pre-requisities for your chat application to support resumable streams:
+
+- Installing the [`resumable-stream`](https://www.npmjs.com/package/resumable-stream) package that helps create and manage the publisher/subscriber mechanism of the streams.
+- A [Redis](https://vercel.com/marketplace/redis) instance to store the stream state.
+- A table that tracks the stream IDs associated with a chat.
+
+To resume a chat stream, you can use the `experimental_resume` function returned by the `useChat` hook. You will typically call this function during the initial mount of the hook.
+
+```
```

Comment:
language

Reply:
updated, but can you verify since it was unclear where to improve